_pytest/config.py

```diff
@@ -916,15 +917,57 @@ def _initini(self, args):
         self._parser.addini('addopts', 'extra command line options', 'args')
         self._parser.addini('minversion', 'minimally required pytest version')
 
+    def _consider_importhook(self, args, entrypoint_name):
+        """Install the PEP 302 import hook if using assertion re-writing.
+
+        Needs to parse the --assert=<mode> option from the commandline
+        and find all the installed plugins to mark them for re-writing
+        by the importhook.
+        """
+        import _pytest.assertion
```

Comment:
The import here is to avoid an circular import between _pytest.config and _pytest.assertion.  I don't like this very much but moving code seems very complex to impossible so this is probably the best compromise.